### DISCUSSION TITLE
VS1 Version possible?

### DISCUSSION THREAD

**Discussion by Wilkware** [philippoo66/optolink-splitter#10] [Ideas]
Ich habe noch eine alte Vitotronic 150 KB1 mit KW2 Protokoll und per Zufall ein Optolink Adapter geerbt.
Mit vcontrold bekomme ich Werte - würde aber viel lieber den Spitter nutzen!

Wäre eine Unterstützung von VS1 denkbar?

VG Heiko

**Comment by philippoo66** [comment]
Moin Heiko!

Danke für's Interesse! Eigentlich sollte auch VS1 kein Problem sein. Ich hatte es ein wenig eilig, deshalb hatte ich mich erstmal auf das 300er Protokoll konzentriert. Wenn du dich als Tester zur Verfügung stellst, bau ich VS1 mal ein.

Grüsse!
Phil

**Comment by Wilkware** [comment]
Moin Phil,

das wäre echt cool! Klar, stehe als Tester bereit! Sag einfach bescheid wenn ich was tun soll.

Danke & Ciao
Heiko

**Reply by philippoo66** [reply]
gleich hab ich erstmal Repair Cafe, mal schaun, vlt komm ich heute abend zum Anfangen

**Reply by philippoo66** [reply]
moin Heiko @Wilkware !

in https://github.com/philippoo66/optolink-splitter/tree/vs1test gibt es den 'initial upload' mit VS1/KW Unterstützung.

ich habe neben optolinkvs2.py ein Modul optolinkvs1.py gemacht und 'mechanisch' gemäss [Markus Fritzes Beschreibung](https://github.com/sarnau/InsideViessmannVitosoft/blob/main/VitosoftCommunication.md#vs1) die Prozeduren darin angepasst, mit einer Ausnahme: ich mache kein F8 Lesen zur Aufrechterhaltung der Kommunikation, sondern den beschriebenen 'reset', wenn länger als 0.5 Sekunden nix passiert ist (im Klartext: es wird dann erstmal auf 0x05 gewartet). Mit Vitoconnect wird dieses die 'Sync Erhaltung' machen, ohne dauert es dann eben ein paar Zehntelsekunden länger, wenn nach einer Pause wieder was kommuniziert werden soll.

Zwischen optolinkvs2 und optolinkvs1 wird entsprechend dem Schalter vs1protocol in der settings_ini ausgewählt.

Ich habe bisher nur in VS Code `F5` gedrückt und dabei gab es keine Fehlermeldung, es ist aber auch gleich wieder ausgestiegen, weil kein serial Port da war. Python wird aber on demand interpretiert, also heisst das 'keine Fehlermeldung' erstmal nicht viel.

Da du dich zum Testen bereit erklärt hast, würde ich dich bitten, jetzt mal den vs1test branch runterzuclonen, die settings_ini ggf noch anzupassen und das Ding mal laufen zu lassen!

Fragen und Rückmeldungen dann hier, Fehlermeldungen bitte mit Kopie des kompletten Outputs.

danke & Grüsse!
Phil

**Reply by philippoo66** [reply]
hm, kannst doch erstmal noch warten, ich hab es mal auf mein reales system geschoben. irgendwas mit der Umschaltung haut wohl noch nicht hin.

**Reply by philippoo66** [reply]
nobody is perfect, noch ein paar andre Sachen, debugging läuft........

**Reply by philippoo66** [reply]
so, ready to go. bei mir läuft es soweit. 

die Connection mit Viessdata haut noch nicht hin, weil das auf 0x05 wartet und dann recht zügig in Timeout geht. Da muss ich noch mal gucken.

Aber ansonsten ist getestet:
- Polling liste mit publish MQTT
- MQTT cmnd read & write
- TCP/IP read & write

'warte gespannt auf deine Rückmeldung.

Vielleicht bau ich auch noch die 'Kommunikationserhaltung' ein, man muss nach sync_elapsed wirklich das Protokoll neu initialisieren. Das Problem ist, dass der Timer in einem eigenen Task läuft und ein tx/rx erstmal nicht mit den andren Tasks synchronisiert wäre und das Durcheinander geben könnte...

**Reply by philippoo66** [reply]
mir fällt grad auf dass da mit Vitoconnect noch viconn_util.detect_vs2() drin ist. Das muss bei VS1 natürlich noch aus/umgeschaltet werden. benutzt du ein Vitoconnect?

**Reply by Wilkware** [reply]
Moin, 

teste nachher gleich mal - bin gespannt!
Ich habe zwar eine Vitoconnect, aber die wollte ja nicht - deshalb hatte ich eine Alternative gesucht und auf Dein Projekt gestoßen :)

Melde mich dann mit den Ergebnissen zurück!

Danke Heiko

**Comment by Wilkware** [comment]
Hallo Phil,

branch vs1test erfolgreich geclont :)

In der optolinkvs1.py ist der Port noch hardcoded auf COM4 => habe ich geändert auf /dev/ttyUSB0

Starten geht dann, aber ich glaub es will noch nicht so richtig => Ausgabe endlos ...

:/opt/optolink# python3 optolinkvs1.py
init_protocol vs1 True
init_protocol vs1 True
R tx 01 f7 00 f8 08
rx 20 92 00 02 ff ff ff ff
0x00f8 20 92 00 02 ff ff ff ff
R tx f7 08 02 02
rx 75 01
KT 75 01 37.3
R tx f7 08 04 02
rx b4 01
WW b4 01 43.6
init_protocol vs1 True
R tx 01 f7 00 f8 08
rx 20 92 00 02 ff ff ff ff
0x00f8 20 92 00 02 ff ff ff ff
R tx f7 08 02 02
rx 75 01
KT 75 01 37.3
R tx f7 08 04 02
rx b4 01
WW b4 01 43.6
init_protocol vs1 True
R tx 01 f7 00 f8 08
rx 20 92 00 02 ff ff ff ff
0x00f8 20 92 00 02 ff ff ff ff
R tx f7 08 02 02
rx 77 01
KT 77 01 37.5
R tx f7 08 04 02
rx a7 01
WW a7 01 42.3

Mach ich was flasch?

Gruß Heiko

**Reply by philippoo66** [reply]
oh, ... die optolinkvs1.py ist nur ein 'Hilfsmodul', das bei gesetzem vs1 Schalter statt der optolinkvs2.py benutzt wird. Die main da drin stammt aus der Zeit, als es noch die optolinkvs2 war, wo ich im Rahmen des Repo py-optolink die vs2 Kommunikation in Python programmiert hatte (fast jedes Hilfsmodul hat eine eigene main zu Testzwecken, die aber im optolink-switch natürlich nie aufgerufen werden). Dass sie jetzt trotzdem funktioniert, zeugt ja schon mal von solider Programmierung ;-) Die main macht genau was sie machen soll, nämlich zyklisch die Gerätekennung, die Kesseltemperatur (37.3°C) und die Speichertemperatur (43.6°C) auslesen. Das 'init_protocol vs1 True' kommt immer nach Ablauf der 0,5s zwischen zwei Kommunikationen. 

Ich habe schon den richtigen Punkt gefunden um das 'keep alive' in den switch zu programmieren, das mach ich später noch. Was ich auch noch machen muss, ist die ganze vicon_util auf vs1 zu erwetern, aber solange du das Vitoconnect nicht benutzt, geht es (so hab ich ja auch getestet). Trotzdem wäre es schön, wenn wir das irgendwann mal testen könnten (wenn es fertig ist).

so, jetzt das Wesentliche: starte den optolink switch bitte mittels
```
python3 optolink_switch.py
``` 

dann wird der switch ausgeführt und nicht nur eine Testfunktion eines Hilfsmoduls ;-)
Grüsse!

**Reply by philippoo66** [reply]
ich hab noch mal n kleines Update gemacht. das mit dem keep_alive wie ich's mir gedacht hatte haut nicht hin wenn 1-wire Sensoren am Start sind, weil die viel zu langsam sind und alles durcheinander bringen. Also im Prinzip gelassen wie gehabt, nur etwas gepimpt.

Für die Sache mit dem Vitoconnect hätte ich gerne vorher eine Aufzeichnung, wie sich verhält bei Init bei KW Protokoll, das würde einiges einfacher machen. Hast du eigentich den CP2102 Adapter?

Was hast du eigentlich vor mit den Daten? Der Optolink-Switch 'switcht' ja nur die Daten, anzeigen und so steht ja auf nem andren Blatt...

**Reply by Wilkware** [reply]
Moin Phil,

ach, da bin ich vor lauter Euphorie wohl übers Ziel hinausgeschossen ;-)
Soo, musste erstmal mqtt Modul auf > 1.6.1 updaten was wieder andere Probleme in anderen Scripts mit sich brachte, aber das läuft jetzt wieder alles rund :)
Was soll ich sagen ... es läuft :) Super - good job!

Habe erstmal mit 4 Variablen angefangen und muss jetzt nach und nach aufrüsten!

Ich habe ein Vitoconnect von meinem Bruder geerbt, der hat ne neue Anlage mit allem drin und drum bekommen, aber braucht dieses Smarte Zeugs nicht :)
Habe dann halt erfolglos versucht die Kiste an meine alte Anlage zum Laufen zu bringen - erfolglos. Er sagt zwar das die Anlage verbunden ist und läuft, aber Daten kommen da gleich NULL ... und es blinkt auch die ganze Zeit gelb. Soweit ich gelesen habe wird eine Vitotronic 150 auch nicht unterstützt - wegen dem alten Protokoll :(
Und nein, ich habe nicht den CP2102 Adapter.

Was mache ich mit den Daten? Ich pumpe sie in meine Haussteurungsanlage IP Symcon (www.ip-sy,con.de). Darüber steuere ich so ziemlich alles was ich "smart" habe. Die Heizung war bis jetzt leider außen vor.
Hatte in letzter Zeit viele Probleme mit der Heizung, viele davon hätte ich schneller erkannt wenn ich schon die Abfrage gehabt hätte (Brennersörung, kaputte Umwälzpumpe, defekter Sensor an der Solartermin usw.) Das will und kann ich jetzt gut überwachen.
Werde nächstes Jahr auch längere Zeit im Ausland sein und da ist es gut zu wissen ob alles rund läuft :)


Gruß Heiko

**Reply by philippoo66** [reply]
Moin Heiko!

'freut mich zu hören, dass es läuft!

Gegen die Geschicht mit den Abhängigkeiten hilft venv - [Virtual Environment](https://docs.python.org/3/library/venv.html) kennst du das? Damit lässt sich das ganze Projekt mit allen Abhängigkeiten in einer Viruellen Umgebung (vlt vergleichbar mit einer Virtuellen Maschine unter Windows) betreiben, und du umgehst alle 'Kreuzabhängigkeiten'.

>  ich habe nicht den CP2102 Adapter.

schade, naja, dann muss ich das mit dem Verbindungs-Init anders lösen und mir 'vorstellen', wie es laufen würde und es demensprechend einbauen... Aber wenn du es eh nicht benutzt, ist das je erstmal nicht so dringend. Ich meine aber, das Vitoconnect unterstützt grundsätzlich auch Steuerungen mit KW Protokoll, aber vielleicht nur ab einem bestimmten Datum oder so.

Wenn du den 'Rückweg' testest, also das Setzen von Werten aus der Hausautomatisierung heraus, probier es vorsichtshalber erstmal mit z.B. Niveau oder Steigung der Heizkennlinie. Ich habe es mal mit Warmwasser-Soll probiert und absichtlich auf einen unzulässigen Wert (150°C) zu setzen versucht, das wurde zwar ordentlich zurückgewiesen, aber danach hatte die Heizung so getan, als hätte ich garkeine WW Bereitung installiert. Das hat mich viele Stunden gekostet, das rauszufinden und wieder grade zu biegen...

Eine Datenpunktliste zu deiner Heizung hast du?

Dann mal viel Spass mit dem neuen Zugriff & danke für's Probieren und die Rückmeldung!

beste Grüsse!
Phil

**Reply by Wilkware** [reply]
Hi Phil,

> Gegen die Geschichte mit den Abhängigkeiten hilft venv
Bin ich jetzt bei der Fehleranalyse drauf gestoßen, kannte ich so noch gar nicht. Wollte aber für die Geschichte den Aufwand nicht treiben. Muss ich mir mal in einer ruhigen Minute reinziehen.

Danke für den Tipp mit der Rückmeldung! Werde vorsichtig sein, ansonsten komm ich zu Dir ;-)

Noch 2 Fragen zur Nutzung:
1) Du schreibst ja noch einiges an Meldungen auf die Konsole - änderst Du das noch?
2) Wie läuft das Script bei Dir (daemon usw.)?

Gruß Heiko

**Reply by Wilkware** [reply]
Nachtrag, was hast Du für eine Anlage? Hinsichtlich DatenPunkte habe ich was. Habe die auch aus dem VissData verglichen - gibts schon unterschiede aber nicht so einfach was nun richtig ist und woher weiß man ob das vorhanden Equipment das auch supported :)

Gruß HEiko

**Reply by philippoo66** [reply]
oki, die debug prints mach ich noch raus und sag dann Bescheid (wahrscheinlich heute abend).

ich starte das als systemd service wie [im wiki beschrieben](https://github.com/philippoo66/optolink-splitter/wiki/optolinkvs2_switch-automatisch-starten) dann gehen die debug Ausgaben eh in's Nichts ;-)

gemäss dem ausgelesenen DeviceIdent (oben) müsste das hier eigentlich die Liste für deine Steuerung sein:
```
0x2092;0x0100;0x0103;;;Vitotronic 150 KB1 Waermebedarfsgefuehrte, digitale Kesselkreisregelung, für 1 Heizkreis ohne Mischer ; V150KB1
```
ich hab die mal angehängt, es kann aber durchaus noch weitere Datenpunkte geben wie [hier ausgeführt](https://github.com/philippoo66/ViessData21?tab=readme-ov-file#dp_listen_2zip) ...

[DP_V150KB1.txt](https://github.com/user-attachments/files/15568902/DP_V150KB1.txt)

**Reply by philippoo66** [reply]
ps. meine Anlage ist eine Vitodens 300-W

debug weg komm ich erst später zu, is schon mitten in der Nacht und ich muss in's Bett

**Reply by philippoo66** [reply]
so, debug prints (der Telegramme) sind raus. du müsstest auch noch `show_opto_rx` in der settings_ini auf False setzen.
